tests/test_project.py

```diff
@@ -324,7 +329,7 @@ def test_context(job):
             with switch_to_directory(project.root_directory()):
                 starting_dir = os.getcwd()
                 with redirect_stderr(StringIO()):
-                    A().run()
+                    A(entrypoint=self.entrypoint).run()
```

Comment:
These changes are obviously very concerning. Why can the entrypoint no longer be determined automatically?

Reply:
It never could. We only detect entrypoint when calling main. That is what we had decided upon.

Reply:
Yes, I remember that conversation and the decision and that's all fine. But just from a high-level point of view, why do we have to change the tests? Like, what is missing that we could execute those before without specifying this and now we can't?